plio/views.py

```diff
@@ -335,6 +350,203 @@ def copy(self, request, uuid):
 
         return Response(self.get_serializer(plio).data)
 
+    @action(
+        methods=["get"],
+        detail=True,
+        permission_classes=[IsAuthenticated, PlioPermission],
+    )
+    def metrics(self, request, uuid):
+        """Returns usage metrics for the plio"""
+        # return 404 if user cannot access the object
+        # else fetch the object
+        plio = self.get_object()
+
+        # no sessions have been created for the plio: return
+        if not Session.objects.filter(plio=plio.id):
+            return Response({})
+
+        import numpy as np
+
+        query = f"""
+            WITH summary AS (
+            SELECT
+                session.id,
+                session.plio_id,
+                session.watch_time,
+                session.retention,
+                ROW_NUMBER() OVER(PARTITION BY session.user_id, session.plio_id
+
+                                ORDER BY session.id DESC) AS rank
+                FROM {connection.schema_name}.session
+            )
+            SELECT id, watch_time, retention
+                FROM summary
+            WHERE rank = 1 AND plio_id = {plio.id}
+        """
+        with connection.cursor() as cursor:
+            cursor.execute(query)
+            results = cursor.fetchall()
+
+        df = pd.DataFrame(results, columns=["id", "watch_time", "retention"])
+
+        # number of unique viewers and average watch time
+        num_views = len(df)
+        average_watch_time = df["watch_time"].mean()
+
+        # retention at one minute
+        if plio.video.duration is None or plio.video.duration < 60:
+            # the metric is not applicable in this case
+            percent_one_minute_retention = None
+        else:
+            # convert "0,1,0" to ["0", "1", "0"]
+            df["retention"] = df["retention"].apply(lambda row: row.split(","))
```

Comment:
Why not move the metrics code to some other file? Will help us as well if we are going to add more metrics, will also keep the `plio/views.py` cleaner

Like there's a file for all the queries, `queries.py` so it makes sense to add these queries there as well